api/backend/duckdb.py

```diff
@@ -22,55 +22,56 @@
             l_comment VARCHAR NOT NULL);"""
 
 
-def ConnectDuckDB():
-    con = duckdb.connect()
-    con.install_extension("substrait")
-    con.load_extension("substrait")
-    con.install_extension("tpch")
-    con.load_extension("tpch")
-    con.execute(query=schema_lineitem)
-    logger.success("DuckDb initialized successfully")
-    return con
+class DuckDBConnection:
+    def __init__(self):
+        self.connection = None
 
+    def connect(self):
+        self.connection = duckdb.connect()
+        self.connection.install_extension("substrait")
+        self.connection.load_extension("substrait")
+        self.connection.install_extension("tpch")
+        self.connection.load_extension("tpch")
+        self.connection.execute(query=schema_lineitem)
+        logger.success("DuckDb initialized successfully")
 
-def CheckDuckDBConnection(con):
-    status = {"db_health": "unavailable"}
-    try:
-        con.execute(query="SHOW TABLES;").fetchall()
-        status["db_health"] = "up and running"
-    except Exception as e:
-        logger.info(
-            "Healthcheck failed for DuckDB"
-            "Initializing new connection object. Details: ",
-            str(e),
-        )
-    finally:
-        return status
+    def check(self):
```

Comment:
It might make more sense to have the separate method for table creation and doing the health check as part of that?  And then store that health status in a class variable?  What do you think?